server/src/main/java/org/elasticsearch/common/util/CuckooFilter.java

```diff
@@ -52,6 +50,10 @@
  * (do not need to waste slots on duplicate fingerprints), and we do not need to worry
  * about inserts "overflowing" a bucket because the same item has been repeated repeatedly
  *
+ * NOTE: this CuckooFilter exposes a number of Expert APIs which assume the caller has
+ * intimate knowledge about how the algorithm works.  It is recommended to avoid these
+ * APIs, or better yet, use {@link SetBackedScalingCuckooFilter} instead.
```

Comment:
I'd just say its recommended to use `SetBackedScalingCuckooFilter` and not touch this class